File: libraries/StatusBarCompat/src/main/java/moe/feng/material/statusbar/NavigationBarFooterView.java
package moe.feng.material.statusbar;

import android.content.Context;
import android.os.Build;
import android.util.AttributeSet;
import android.view.View;
import android.view.ViewGroup;

import moe.feng.material.statusbar.util.ViewHelper;

public class NavigationBarFooterView extends View {

	public NavigationBarFooterView(Context context) {
		this(context, null);
	}

	public NavigationBarFooterView(Context context, AttributeSet attrs) {
		this(context, attrs, 0);
	}

	public NavigationBarFooterView(Context context, AttributeSet attrs, int defStyle) {
		super(context, attrs, defStyle);
	}

	@Override
	public void onMeasure(int widthSpec, int heightSpec) {
		super.onMeasure(widthSpec, heightSpec);
		adjustHeight();
	}

	@Override
	public void invalidate() {
		super.invalidate();
		adjustHeight();
	}

	public void adjustHeight() {
		ViewGroup.LayoutParams params = getLayoutParams();
		params.height = Build.VERSION.SDK_INT >= Build.VERSION_CODES.KITKAT ? ViewHelper.getNavigationBarHeight(getContext()) : 0;
	}

}

File: libraries/StatusBarCompat/src/main/java/moe/feng/material/statusbar/NavigationBarFooterView.kt
package moe.feng.material.statusbar

import android.content.Context
import android.os.Build
import android.util.AttributeSet
import android.view.View

import moe.feng.material.statusbar.util.ViewHelper

class NavigationBarFooterView @JvmOverloads constructor(context: Context, attrs: AttributeSet? = null, defStyle: Int = 0) : View(context, attrs, defStyle) {

	public override fun onMeasure(widthSpec: Int, heightSpec: Int) {
		super.onMeasure(widthSpec, heightSpec)
		adjustHeight()
	}

	override fun invalidate() {
		super.invalidate()
		adjustHeight()
	}

	fun adjustHeight() {
		val params = layoutParams
		params.height = if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.KITKAT) ViewHelper.getNavigationBarHeight(context) else 0
	}

}
